baldrick/github/github_api.py

```diff
@@ -542,14 +573,35 @@ def set_check(self, name, summary, commit_hash='head', details_url=None,
         tt = datetime.utcnow()
         completed_at = tt.isoformat(timespec='seconds') + 'Z'
 
-        output = {'title': name, 'summary': summary}
-        parameters = {'name': name, 'head_sha': commit_hash, 'status': status,
-                      'conclusion': conclusion, 'completed_at': completed_at,
-                      'output': output}
+        # If name isn't specified revert to external_id
+        name = name or f"{current_app.bot_username}:{external_id}"
+
+        output = {'title': title, 'summary': summary or ''}
+        if text is not None:
+            output['text'] = text
+
+        parameters = {'external_id': external_id, 'name': name, 'head_sha':
+                      commit_hash, 'status': status, 'output': output }
+
         if details_url is not None:
             parameters['details_url'] = details_url
 
-        response = requests.post(url, headers=headers, json=parameters)
+        if status == "completed" and conclusion is None:
+            logger.warning(
+                "When a GitHub check status is completed, conclusion must be specified, setting it to 'neutral'")
+            conclusion = "neutral"
+
+        if conclusion is not None:
+            parameters['conclusion'] = conclusion
+            parameters['completed_at'] = completed_at
+            parameters['status'] = "completed"
```

Comment:
@pllim This is the fix to the issue I was describing to you in chat. I think this is an ok compromise, because if you set conclusion you probably mean the job to be finished?

Reply:
On further inspection of the GitHub API, GitHub will do this if we don't:

> Note: Providing `conclusion` will automatically set the `status` parameter to `completed`.